### ISSUE TITLE
2024.06 Release Tracking

### ISSUE THREAD

**Issue by rtsisyk** [organicmaps/organicmaps#8252]
To improve the overall visibility of processes within the organization, we're shifting the release procedure from Telegram chat to GitHub for the June release. The procedure itself has been updated based on comments from the previous release:

1. Added step-by-step checklist
2. Added sign-off from interested parties.
3. Added dedicated release branch.
4. Added What's New into plan.

The new procedure is experimental. We will learn from this experience and make adjustments accordingly.

---------------------------------------------------------

Expected Store Submission: Sun 2 Jun (around 20:00 GMT)
Expected Full Rollout to Users: Fri 7, Jun
Release Manager: @rtsisyk (piloting the new experimental procedure)

- [x] The last call for any new PRs to be merged into the master branch | @rtsisyk | Mon, May 27
- [x] [No release blockers](https://github.com/organicmaps/organicmaps/milestone/6)  | @rtsisyk | Mon, May 27
- [x] The new data generator is started | @vng | Wed, May 29
- [x] A draft of release notes is prepared in etherpad | @pastk  | Wed, May 29
- [x] The master branch looks healthy | @rtsisyk | Sat 1 Jun
 	- [x] Android is tested by the monkey (15 minutes) | @rtsisyk
 	- [x] Android is tested on a device (15 minutes) | @rtsisyk
 	- [x] iOS is tested on a device (15 minutes) | @rtsisyk
- [x] A new release branch is created from the master  | Sat 1 Jun
 	- [x] GitHub Branch Protection is configured
 	- [x] GitHub Environments are configured 
- [x] A PR with the new data is raised | @vng | Sat 1 Jun 
- [x] A PR with the new release notes is raised |  ? @pastk | Sat 1 Jun
- [x] The release-candidate sign-off | Sun, Jun 02
 	- [x] Android is tested by monkey | @rtsisyk
 	- [x] Android is tested manually by @Jean-BaptisteC
 	- [x] Android is tested by @rtsisyk
 	- [x] Android is tested by @vng
 	- [x] iOS is tested by @kirylkaveryn 	 
 	- [x] iOS is tested by @rtsisyk
 	- [ ] iOS is tested by @vng
- [x] The release-candidate is published | @rtsisyk | Sun, Jun 02
 	- [x] The PR with new data is merged into the master
 	- [x] The PR with release notes is merged into the master
 	- [x] Android RC is published on GitHub as a pre-release
 	- [x] Android RC is submitted to Google Play Closed Beta for review
 	- [x] Android RC is published on Firebase AppTester for testing
 	- [x] Android RC is published on Telegram for testing
 	- [x] F-Droid build is triggered | @pastk and @rtsisyk to figure out how to check
 	- [x] iOS RC is submitted to Apple TestFlight for review
 	- [x] Data is synchronized to all servers
 	- [x] Metaserver is updated to include new data version
- [x] iOS RC TestFlight is approved | @rtsisyk | Tue, Jun 04 – Wed, Jun 05
 	- [x] iOS RC AppStore metadata is updated | @yanatalia | The call on Jun 03 
 	- [x]  iOS RC TestFlight link is published on Telegram for testing
 	- [x] iOS RC is sent for review to App Store production with manual release
- [x] Android RC approved for Google Play Closed Beta | @rtsisyk |  <s>Tue, Jun 04–Wed, Jun 05</s>Mon, Jun 01
 	- [x] Android RC is sent for review to Google Play Production with manual release | @rtsisyk
- [x] AppStore ~7~ 3 days phased rollout | @rtsisyk | Wed, Jun 05
- [x] Google Play 10% phased rollout | @rtsisyk | <s>Wed, Jun 05</s>Tue, Jun 04
- [x] Metaserver is updated to include new app versions  | @rtsisyk | Wed, Jun 05
- [x] Google Play 20% phased rollout | @rtsisyk | Thu, Jun 06
- [x] The release final sign-off | Fri, Jun 07
 	- [x] Firebase reports by @rtsisyk
 	- [x] Google Play reports by @rtsisyk
 	- [x] TestFlight reports by @kirylkaveryn
 	- [x] Email & Telegram reports by @biodranik
- [x] AppStore 100% rollout | @rtsisyk | Fri, Jun 07
- [x] Google Play 100% rollout | @rtsisyk | Fri, Jun 07
- [ ] App Gallery 100% rollout | @rtsisyk | Fri, Jun 07
- [x] GitHub - remove "pre-release" and mark the release as "latest" | @rtsisyk | Fri, Jun 07
- [x] F-Droid check | @pastk | Fri, Jun 07
- [x] Announcements
  	- [x] Telegram announcement | @biodranik 
  	- [ ] Mastodon announcement | @biodranik
  	- [ ] Facebook announcement | @biodranik 
  	- [x] Website announcement | @biodranik

**Comment by pastk** [comment]
> * A PR with the new release notes is raised |  ? @pastk | Sat 1 Jun

I'll be offline (or very limited availability) around these dates, sorry..

I'll try to keep relnotes updated in etherpad before then.

Also could we use the RC testing time to prep translations of the relnotes? And then update them in stores just before promoting RC to release? Is it technically possible / easy / feasible?

**Comment by pastk** [comment]
> Yes, Google Play can be updated at ANY time.

From https://github.com/organicmaps/organicmaps/pull/8083#issuecomment-2094827524 I didn't understand:
does it require editing a special [google-play-translations.txt](https://github.com/organicmaps/organicmaps/files/15213279/google-play-translations.txt) file or we could just update usual files and then use a plugin?

> AppStore should be updated before sending the release to review.

Is it when RC is pushed on June 2nd?
Or when " iOS RC is sent for review to App Store production with manual release" around June 4-5th?

**Comment by pastk** [comment]
Just a note that with this workflow a release will contain map data outdated by 9 days already (release on 7th of Jun, data is from 29th of May).

The minimum gap we can get to is ~6 days (3 days for planet generation + 3 days for Apple/Google review).
Its possible to generate a fresher map data while RC is being tested and then use it to push the prod version (which will have to be sent to a separate review). But probably it doesn't make sense (for now at least) to make the release workflow more complicated to "win ~3 days of freshness".

Whats great about this workflow is that keeping each release in its own branch will allow for relatively fast and easy data-only releases.

**Comment by pastk** [comment]
>     3. Send the same bundle for the review to AppStore prod.
>        
>        * This requires final localized Releases Notes that can't be changed later. AppStore is not such flexible as Google Play and Release Notes can only be changed during the review of a new version.

Ok great, then right after the EN relnotes version is approved & merged on June 2 we can ask the translation team to provide translations till June 4-5 and merge an update. I understood the translations will be picked automagically when a prod iOS build will be sent for a review.

To update translations in GPlay we'll need a simple script that'll produce a special format [google-play-translations.txt](https://github.com/organicmaps/organicmaps/files/15213279/google-play-translations.txt) file by aggregating separate files like `android/app/src/fdroid/play/listings/en-US/release-notes.txt`.

Any volunteers to write it?
(unfortunately I don't have time atm)

**Comment by brdbkr** [comment]
Can the next update remove Kayak integration please?

**Comment by pastk** [comment]
New data gen should be started off the new release branch (and hence after it had been created).

**Comment by kirylkaveryn** [comment]
@rtsisyk seems like it's Ok.
Also I've  made hot fix for this opening hours issue:

https://github.com/organicmaps/organicmaps/pull/8371

**Comment by vng** [comment]
@rtsisyk
- I don't have data generation plan or schedule :)
- Do you want me to put new data generation?

**Comment by Arthur-GYT** [comment]
F-droid RC version is published ? And on what repository ?

**Comment by pastk** [comment]
> F-droid RC version is published ? And on which repository ?

It has detected a new version already, but as mentioned above it'll take some days to build and publish. You can track current status via the F-Droid Build Status app.

**Comment by pastk** [comment]
> To update translations in GPlay we'll need a simple script that'll produce a special format [google-play-translations.txt](https://github.com/organicmaps/organicmaps/files/15213279/google-play-translations.txt) file by aggregating separate files like `android/app/src/fdroid/play/listings/en-US/release-notes.txt`.

- https://github.com/organicmaps/organicmaps/pull/8381

@rtsisyk could you please upload the final relnotes version?

**Comment by pastk** [comment]
7-days phased iOS roll-out seems too long, no? Is there an option to shorten it?

**Comment by pastk** [comment]
F-Droid release is fully done.

**Comment by biodranik** [comment]
iOS rollout does not go like on Android. There are some crash reports in Testflight, but they don't look new at first sight. No other reports yet. The main source of reports is usually a publicly announced/available beta version that matches the release version.

I've pushed iOS to 100%

**Comment by kirylkaveryn** [comment]
> @kirylkaveryn could you please check appstore & test flight crash reports for regressions?

We have a new one crash. Seems like in the core:
https://github.com/organicmaps/organicmaps/issues/8403

**Comment by pastk** [comment]
> What is about making a data-only release in ~2 weeks, between Jun 17–23? @pastk @vng

It'd be great to try it!

**Comment by pastk** [comment]
Need to add more items into the check list:
- [ ] publish release announcement on social media
- [ ] publish release announcement on the web-site https://organicmaps.app/news/

**Comment by vng** [comment]
> I am proposing merging the https://github.com/organicmaps/organicmaps/tree/release/2024.06.02 release branch into master

So? It is already a blocker for my work. We should add this step into the Release process.

**Comment by pastk** [comment]
> @vng , could you please start the generator on this week to get data for the next week?

Just a reminder that it should be started from the `release/2024.06.02` branch :pray:

**Comment by pastk** [comment]
@biodranik what commits should be added into the upcoming hotfix/data release?

**Comment by biodranik** [comment]
@pastk 97a0544a55636ffc4a696d041a8c3c9063f02835 for iOS hotfix

**Comment by pastk** [comment]
Or shall we create a separate branch for the minor (fix & data) release?
And merge #8468, #8471, https://github.com/organicmaps/organicmaps/commit/97a0544a55636ffc4a696d041a8c3c9063f02835 in there?

**Comment by biodranik** [comment]
Requrements are:
1. A branch for stable releases + hotfixes
2. Having the last actual world map + fixed tests in the master as soon as it is possible/feasible.
3. Avoid freezing the master branch to unblock code reviews, speed up the development and testing, and motivate contributors.

**Comment by pastk** [comment]
Awesome, looking forward for the first data-only release ever! 🎉

**Comment by biodranik** [comment]
Usually when publishing releases, we always also publish the corresponding version as beta to get preliminary feedback from users. @rtsisyk can you please add this step to data-only releases too?

Many beta testers are only using beta versions, and it is (and already was) very helpful to get their feedback about any found issues.

**Comment by pastk** [comment]
Data-only releases have nothing to beta test (well except the new data).

So if we push data-only releases to beta channels then they might overwrite recent betas that still have something to test (compared to a last release).

**Comment by biodranik** [comment]
Am I hearing you correctly that there is no value in beta testing and it should be avoided because it is too complex to publish betas?

1. This data only release for ios also contains hotfix. Future releases also may have hotfixes. 
2. Data can be broken, as you mentioned. It should also be tested. 
3. Discovering any other issues (not only regressions caused by latest changes) is also important. 
4. We and our contributors should also use betas when possible, to discover any issues ASAP
5. Publishing betas manually now takes around 15 minutes, if beta release notes are prepared. This time can be reduced by automating the process.

**Comment by pastk** [comment]
> if we push data-only releases to beta channels then they might overwrite recent betas that still have something to test (compared to a last release)

Thats my point.
Overwriting with data-only releases will diminish testing of new unreleased changes, which is the main value of betas IMHO.

Of course we can update the latest beta (that contain unreleased features) with data and hotfix changes, but that seems like too much overhead (create a separate branch for each beta and then merge data changes in there?)

Sorry but I don't see value in your points of beta-testing data-only/hotfix releases. Because those changes are being "tested" (used) by all stable release version users anyway.

**Comment by pastk** [comment]
Perhaps we should clarify also what is a "hotfix" and what to include into interim data releases.

In my view a "hotfix" is something so critical/important that we choose to cut the corners (incl. skip beta-testing) to release it ASAP. It means that the hotfix value outweights the risk of new bugs it could possibly introduce. ASAP also means that it shouldn't wait for a new data.

If a bugfix is too risky to be released without beta-testing and can wait for that extra testing time, but still is so important we want to push it before the next full-blown release, then a new beta with that bugfix should be formed sometime before the interim data release.

If a bugfix is of high importance (but not as critical as a "hotfix") and is straightforward / very low risk, then we can just include it into upcoming interim data release.

**Comment by pastk** [comment]
> Of course we can update the latest beta (that contain unreleased features) with data and hotfix changes, but that seems like too much overhead (create a separate branch for each beta and then merge data changes in there?)

E.g. if we just
- merge the interim release changes into the current `beta` branch
- trigger beta build actions on GH

Would it be enough to update the current beta?
What beta notes testers will see?
Does it make sense at all?

**Comment by pastk** [comment]
> We can continue practicing phased rollout even for data-only releases.

What is the benefit?
Additional testing (data errors, bugs in bug fixes)?

Downsides?
Longer release times and more stale data?
Will it overwrite the recent beta (and unreleased changes in there)?

**Comment by pastk** [comment]
Please add app ids to your list :pray:

**Comment by pastk** [comment]
> The next time we can push the update to testflight also.

Why not this time if its easy?

**Comment by biodranik** [comment]
@Jean-BaptisteC mentioned that screenshots were not updated in GP release. Can it be fixed?

Producing alphas/betas for testers should be included in the release procedure.
